ElenHakobjanyan/Words Frequencies

```diff
@@ -0,0 +1,86 @@
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <string>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+bool addWordsToMap(string, map<string, int>&, int&);
+void sortAsVec(map<string, int>&, vector<pair<string, int>>&);
+void print(vector<pair<string, int>>&, int, int&);
+
+int main()
+{
+	int count = 0;
+	map<string, int> mp;
+	addWordsToMap("war+peace.txt", mp, count);
+
+	vector<pair<string, int>> vec;
+	sortAsVec(mp, vec);
+
+	print(vec, 100, count);	
+	return 0;
+}
+
+bool addWordsToMap(string fname, map<string, int>& mp, int& count)
+{
+	ifstream fstream(fname);
+
+	if (!fstream.is_open())
+	{
+		cerr << "Unable to open '" << fname << "' file\n";
+		return false;
+	}
+
+	string buff;
+	char n;
+
+	while (fstream.get(n))
+	{
+		n = tolower(n);
+		if (n >= 'a' && n <= 'z') buff += n;
+		else
+		{
+			if (buff != "")
+			{
+				//cout << buff << " ";
+				mp.insert_or_assign(buff, ++mp[buff]);
```

Comment:
Would be easier to just use `++mp[buf]`

Reply:
i didn't understand instead of what to use that, what if the word, that's in the buffer isn't in the map yet?

Reply:
It will create a new pair like this - <hello, 0>, then return the 0 which is then being incremented. Take a look at https://godbolt.org/z/P8r6eaGz7

Reply:
so, mp[buff] will make new, if it doesn't find any, right? and the ++ will change the value, which is 0 by default

Reply:
exactly. In some special situations, you have to use `insert_or_assign` but here the good old operator[] will do